ENHANCED_PERMISSIONS_EXAMPLE.md

```diff
@@ -0,0 +1,178 @@
+
+```typescript
+import { usePermissions } from './hooks/usePermissions'
+import { DEFAULT_QUEUE_RULES } from './hooks/usePermissions/permission_rules'
+
+// Example: Document Management System
+const documentRules = [
+  {
+    role: "owner",
+    resource: "document",
+    actions: ["view", "edit", "delete", "share"],
+    condition: (ctx) => ctx.data?.ownerId === ctx.userId
+  },
+  {
+    role: "collaborator",
+    resource: "document",
+    actions: ["view", "edit"],
+    condition: (ctx) => ctx.data?.collaborators?.includes(ctx.userId)
+  },
+  { role: "viewer", resource: "document", actions: ["view"] },
+] as const
+
+function DocumentEditor() {
+  const { results, hasPermission, hasAnyPermission } = usePermissions({
+    userRole: "owner", // Type-safe: "owner" | "collaborator" | "viewer"
+    context: { userId: "user-123" },
+    rules: documentRules,
+    checks: [
+      {
+        resource: "document",
+        action: "edit",
+        data: { ownerId: "user-123", collaborators: ["user-456"] }
+      },
+      {
+        resource: "document",
+        action: "delete",
+        data: { ownerId: "user-123" }
+      }
+    ]
+  })
+
+  // Type-safe helper functions
+  const canEdit = hasPermission("document", "edit")
+  const canManage = hasAnyPermission("document", ["delete", "share"])
+
+  return (
+    <div>
+      {canEdit && <EditButton />}
+      {canManage && <ManageButton />}
+
+      {/* Or use results directly */}
+      {results.map(result => (
+        <div key={`${result.resource}-${result.action}`}>
+          {result.resource} {result.action}: {result.hasPermission ? "✅" : "❌"}
+        </div>
+      ))}
+    </div>
+  )
+}
+```
+
+## Benefits
+
+### For Developers
+- **Type Safety**: Compile-time checking of roles, actions, and resources
+- **Intellisense**: Full autocomplete support in IDEs
+- **Flexibility**: Support any field naming convention
+- **Performance**: Batch permission checking reduces evaluation overhead
+
+### For Projects
+- **Reusability**: Same hook works across different permission systems
+- **Maintainability**: Types automatically stay in sync with rules
+- **Scalability**: Multi-resource checking prevents n+1 permission queries
+
+### For Teams
+- **Consistency**: Standardized permission patterns across the codebase
+- **Documentation**: Types serve as living documentation
+- **Refactoring**: Safe renaming and restructuring with TypeScript
+
+## Migration Path
+
+For existing code using the old API:
+
+```typescript
+// Old API (still works via legacy support)
+const { permissions, hasPermission } = usePermissions({
+  userId: "user-123",
+  role: "staff",
+  resource: "appointment",
+  data: { assignedTo: "user-123" }
+})
+
+// New API (recommended)
+const { results, hasPermission } = usePermissions({
+  userRole: "staff",
+  context: { userId: "user-123" },
+  rules: DEFAULT_QUEUE_RULES,
+  checks: [
+    { resource: "appointment", action: "view", data: { assignedTo: "user-123" } },
```

Comment:
You should just be able to provide resource and data and it should return which actions are allowed